### DISCUSSION TITLE
ZR+ low and high power simulation query

### DISCUSSION THREAD

**Discussion by SaiBhyri** [Telecominfraproject/oopt-gnpy#500] [Q&A] [answered]
Hi Team,

I am simulating ZR+ 0dBm and -10dBm variants.
AFAIK, there is no "output_power parameter" for transceivers.
So, which existing parameter should i tweak to see the change due to output pwr difference? Is it RAODM "target_pch_out_db" parameter?

Thanks,
Sai

**Comment by EstherLerouzic** [comment, marked as answer]
dear @SaiBhyri 

The feature that will correctly handle transceiver output power and min roadm loss is under review here: https://review.gerrithub.io/c/Telecominfraproject/oopt-gnpy/+/1177986

in the meantime you may tweak using delta_pdb in --spectrum option with gnpy-transmission-example script or with "equalization_offset_db" in transceivers for gnpy-path-request script.

Note that you need to compute the expected power offset out or ROADM for your low power channel using the min loss of the ROADM. For example, if your channel  should be equalized at -20dBm out of ROADM, but your ROADM loss is minimum -15dBm, you should use min(-20, -10 -15) = -25dBm to determine the expected equalization for this channel and then report the difference vs normal channels in the delta_pdb = -5dB:

```

{
      "spectrum":[
            {
            "f_min": 191.35e12,
            "f_max":193.1e12,
            "baud_rate": 32e9,
            "slot_width": 50e9,
            "power_dbm": 0,
            "roll_off": 0.15,
            "tx_osnr": 40
            },
            {
            "f_min": 193.15e12,
            "f_max":193.15e12,
            "baud_rate": 32e9,
            "slot_width": 50e9,
            "power_dbm": 0,
            "roll_off": 0.15,
            "tx_osnr": 40,
            "delta_pdb": -5
            },
            {
            "f_min": 193.2e12,
            "f_max":195.1e12,
            "baud_rate": 32e9,
            "slot_width": 50e9,
            "power_dbm": 0,
            "roll_off": 0.15,
            "tx_osnr": 40
            }
      ]
}
```
you can do exactly the same thing usind transceivers "equalization_offset_db" with the gnpy-path-request script.

Note that in both cases this power offset will be repeated at each crossed ROADM, which is probably not what you want.

**Reply by akhil-mutike** [reply]
just a small corecction...min(-20, -10 -15) is not-25dBm.It should be -20dbm
